tests/unit/client.py

```diff
@@ -0,0 +1,301 @@
+from haas.model import db
+from haas.test_common import *
+from haas.network_allocator import get_network_allocator
+import pytest
+import json
+
+import requests
+import os
+import tempfile
+import subprocess
+from subprocess import check_call, Popen
+from urlparse import urljoin
+import requests
+from requests.exceptions import ConnectionError
+#from haas.client_lib.client_lib import hilClientLib
+## Hook to the client library
+from haas.client.base import ClientBase
+from haas.client.auth import auth_db
+from haas.client.client import Client
+
+
+
+
+MOCK_SWITCH_TYPE = 'http://schema.massopencloud.org/haas/v0/switches/mock'
+OBM_TYPE_MOCK = 'http://schema.massopencloud.org/haas/v0/obm/mock'
+OBM_TYPE_IPMI = 'http://schema.massopencloud.org/haas/v0/obm/ipmi'
+
+ep = "http://127.0.0.1:8888" or os.environ.get('HAAS_ENDPOINT')
+username = "hil_user" or os.environ.get('HAAS_USERNAME')
+password = "hil_pass1234" or os.environ.get('HAAS_PASSWORD')
+
+
+auth = auth_db(username, password)
+
+C = Client(ep, auth) #Initializing client library
+
+
+## Following tests check if the client library is initialized correctly
+
+def test_auth_db():
+    auth = auth_db(username, password)
+    assert auth.decode('base64', 'strict') == (":").join([username, password])
+
+
+class Test_ClientBase:
+    """ When the username, password is not defined
+    It should raise a LookupError
+    """
+
+    def test_init_error(self):
+        try:
+            x = ClientBase()
+        except LookupError:
+            assert True
+
+    def test_correct_init(self):
+        x = ClientBase(ep, 'some_base64_string')
+        assert x.endpoint == "http://127.0.0.1:8888"
+        assert x.auth == "some_base64_string"
+
+    def test_object_url(self):
+        x = ClientBase(ep, 'some_base64_string')
+        y = x.object_url('abc', '123', 'xy23z')
+        assert y == 'http://127.0.0.1:8888/abc/123/xy23z'
+
+#For testing the client library we need a running HIL server, with dummy
+#objects populated. Following classes accomplish that end. 
+# It shall:
+#       1. Configures haas.cfg
+#       2. Instantiates a database
+#       3. Starts a server on an arbitary port
+#       4. Populates haas with dummy objects
+#       5. tears down the setup in a clean fashion.
+
+
+#pytest.fixture(scope="module")
+
+def make_config():
+    """ This function creates haas.cfg with desired options
+    and writes to a temporary directory.
+    It returns a tuple where (tmpdir, cwd) = ('location of haas.cfg', 'pwdd')
+    """
+    tmpdir = tempfile.mkdtemp()
+    cwd = os.getcwd()
+    os.chdir(tmpdir)
+    with open('haas.cfg', 'w') as f:
+        config = '\n'.join([
+            '[general]'
+            'log_level = debug',
+
+            '[auth]',
+            'require_authentication = False',
+
+            '[headnode]',
+            'base_imgs = base-headnode, img1, img2, img3, img4',
+            '[database]',
+            'uri = sqlite:///%s/haas.db' % tmpdir,
```

Comment:
I don't think that would work, since you call the cli just to initialize the db, and then again to start the server. If you used an in-memory db you'd lose it when the first process exits.